server/ast/set_var.go

```diff
@@ -27,6 +27,394 @@ func nodeSetVar(node *tree.SetVar) (vitess.Statement, error) {
 	if node == nil {
 		return nil, nil
 	}
-	//TODO: For some reason, the parser does not actually implement SET, so we need to add that
-	return nil, fmt.Errorf("SET is not yet supported")
+	if _, ok := pgCfgVars[node.Name]; !ok {
+		return nil, fmt.Errorf(`ERROR: syntax error at or near "%s"'`, node.Name)
+	}
+	if node.IsLocal {
+		// TODO: takes effect for only the current transaction rather than the current session.
+		return nil, fmt.Errorf("SET LOCAL is not yet supported")
+	}
+	expr, err := nodeExpr(node.Values[0])
+	if err != nil {
+		return nil, err
+	}
+	setStmt := &vitess.Set{
+		Exprs: vitess.SetVarExprs{&vitess.SetVarExpr{
+			Scope: vitess.SetScope_Global, // TODO: always SESSION scope until we support LOCAL scope BUT postgres SESSION scope means it's GLOBAL for SystemVariableRegistry interface
+			Name: &vitess.ColName{
+				Name: vitess.NewColIdent(node.Name),
+				Qualifier: vitess.TableName{
+					Name:      vitess.NewTableIdent("postgres"),
+					Qualifier: vitess.NewTableIdent(""),
+				},
+			},
+			Expr: expr,
+		}},
+	}
+	return setStmt, nil
+}
+
+// pgCfgVars is a list of configuration parameters that can be used in SET statement.
+// All parameters are in lower-case.
+var pgCfgVars = map[string]struct{}{
```

Comment:
You can add a function in the `config` package that does this check against the `postgresConfigVariables` map. That way we have all of the names in one map, rather than remembering to update this map if we update the other map.